File: app/src/main/java/com/example/schedule/ui/PreferencesActivity.java
package com.example.schedule.ui;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.CheckBoxPreference;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.SwitchPreference;

import com.example.schedule.R;
import com.example.schedule.ScheduleApplication;
import com.example.schedule.logic.UpdateHelper;
import com.example.schedule.util.LoadDialog;

import static com.example.schedule.ScheduleApplication.currentTheme;

public class PreferencesActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setTheme(currentTheme);
        setTitle("Настройки");

        // Display the fragment as the main content.
        getSupportFragmentManager().beginTransaction()
                .replace(android.R.id.content, new SettingsFragment())
                .commit();
    }

    public static class SettingsFragment extends PreferenceFragmentCompat {

        LoadDialog loadDialog;

        @Override
        public void onCreate(@Nullable Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);

            loadDialog = new LoadDialog(getFragmentManager());
        }

        @Override
        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {

            // Load the preferences from an XML resource
            addPreferencesFromResource(R.xml.preferences);

            SwitchPreference themePref = findPreference("theme_pref");
            themePref.setOnPreferenceChangeListener((preference, newValue) -> {
                new ScheduleApplication.RestartAppTask().execute();
                return true;
            });

            Preference aboutAppPref = findPreference("about_app_pref");
            aboutAppPref.setOnPreferenceClickListener(preference -> {
                startActivity(AboutAppActivity.newIntent(getContext()));
                return true;
            });

            Preference updateAppPref = findPreference("check_update_app_pref");
            updateAppPref.setOnPreferenceClickListener(pref -> {
                UpdateHelper updateHelper = new UpdateHelper(getContext(), getFragmentManager());
                updateHelper.checkUpdate();
                return true;
            });

        }
    }

    public static Intent newIntent(Context context) {
        return new Intent(context, PreferencesActivity.class);
    }
}

File: app/src/main/java/com/example/schedule/ui/PreferencesActivity.kt
package com.example.schedule.ui

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import androidx.preference.SwitchPreference
import com.example.schedule.R
import com.example.schedule.ScheduleApplication
import com.example.schedule.ScheduleApplication.RestartAppTask
import com.example.schedule.logic.UpdateHelper
import com.example.schedule.util.LoadDialog

class PreferencesActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTheme(ScheduleApplication.currentTheme)
        title = "Настройки"

        supportFragmentManager.beginTransaction()
                .replace(android.R.id.content, SettingsFragment())
                .commit()
    }

    class SettingsFragment: PreferenceFragmentCompat() {

        private lateinit var loadDialog: LoadDialog

        override fun onCreate(savedInstanceState: Bundle?) {
            super.onCreate(savedInstanceState)
            loadDialog = LoadDialog(fragmentManager!!)
        }

        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) { // Load the preferences from an XML resource
            addPreferencesFromResource(R.xml.preferences)

            val themePref = findPreference<SwitchPreference>("theme_pref")
            themePref!!.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { _, _ ->
                RestartAppTask().execute()
                true
            }

            val aboutAppPref = findPreference<Preference>("about_app_pref")
            aboutAppPref!!.onPreferenceClickListener = Preference.OnPreferenceClickListener {
                startActivity(Intent(context, AboutAppActivity::class.java))
                true
            }

            val updateAppPref = findPreference<Preference>("check_update_app_pref")
            updateAppPref!!.onPreferenceClickListener = Preference.OnPreferenceClickListener {
                val updateHelper = UpdateHelper(context, fragmentManager)
                updateHelper.checkUpdate()
                true
            }
        }
    }
}